packages/next/src/shared/lib/router/utils/resolve-rewrites.ts

```diff
@@ -20,14 +24,14 @@ export default function resolveRewrites(
   locales?: readonly string[]
 ): {
   matchedPage: boolean
-  parsedAs: ParsedRelativeUrl
+  parsedAs: ParsedAs
   asPath: string
   resolvedHref?: string
   externalDest?: boolean
 } {
   let matchedPage = false
   let externalDest = false
-  let parsedAs = parseRelativeUrl(asPath)
+  let parsedAs: ParsedAs = parseRelativeUrl(asPath)
```

Comment:
We later assign the return value of `prepareDestination` which just happened to structurally match `ParsedRelativeUrl`. Now it no longer does because `ParsedRelativeUrl ` has `undefined` slashes while `ParsedUrl` has `boolean` slashes.